src/data/ellpack_page.cu

```diff
@@ -185,10 +185,7 @@ struct TupleScanOp {
 
 // Change the value type of thrust discard iterator so we can use it with cub
 template <typename T>
-class TypedDiscard : public thrust::discard_iterator<T> {
- public:
-  using value_type = T;  // NOLINT
-};
+using TypedDiscard = thrust::discard_iterator<T>;
```

Comment:
The purpose of this class is to change the value type away from void, in order to work with cub/thrust in certain edge cases. If it is no longer necessary, maybe cub/thrust has changed upstream? In which case we don't need the using statement we can just use the discard director directly.

Reply:
Maybe we can make the change when we drop cuda 11.2 support.  But at the moment we still need it.